src/server/pkg/obj/obj.go

```diff
@@ -314,3 +315,13 @@ func isNetRetryable(err error) bool {
 	netErr, ok := err.(net.Error)
 	return ok && netErr.Temporary()
 }
+
+// TestIsNotExist is a defensive method for checking to make sure IsNotExist is
+// satisfying its semantics.
+func TestIsNotExist(c Client) error {
+	_, err := c.Reader(uuid.NewWithoutDashes(), 0, 0)
```

Comment:
Not even close to defensive enough, the good news though is that in order for  this to happen it requires a user not just writing uuid values (without dashes) to the top level of their bucket but writing the correct uuid value, so they need to win the uuid lottery. And if this does happen then the server will just restart and role a new uuid.... if someone has written all the uuids to their bucket I expect pachyderm to have some issues though.